src/Aspire.Cli/DotNetCliRunner.cs

```diff
@@ -15,12 +15,22 @@ namespace Aspire.Cli;
 
 internal sealed class DotNetCliRunner(ILogger<DotNetCliRunner> logger, IServiceProvider serviceProvider)
 {
+    private readonly ActivitySource _activitySource = new ActivitySource(nameof(Aspire.Cli.DotNetCliRunner));
+
     internal Func<int> GetCurrentProcessId { get; set; } = () => Environment.ProcessId;
 
-    public async Task<int> RunAsync(FileInfo projectFile, bool watch, string[] args, IDictionary<string, string>? env, TaskCompletionSource<AppHostBackchannel>? backchannelCompletionSource, CancellationToken cancellationToken)
+    public async Task<int> RunAsync(FileInfo projectFile, bool watch, bool noBuild, string[] args, IDictionary<string, string>? env, TaskCompletionSource<AppHostBackchannel>? backchannelCompletionSource, CancellationToken cancellationToken)
     {
+        using var activity = _activitySource.StartActivity(nameof(RunAsync), ActivityKind.Client);
+
+        if (watch && noBuild)
+        {
+            throw new InvalidOperationException("Cannot use --watch and --no-build at the same time.");
```

Comment:
Curious why?

Reply:
Why do this? So we can isolate the timings for build vs. just running. Allows us to zero in on how long it takes for a backchannel to connect.

Reply:
I meant more from a product pov.

Reply:
Its _mostly_ for my own diagnostic purposes. However, I think it might also make for more obvious failure states as well where we can conclusively say that `dotnet run` failed because the build failed vs. the apphost crashed after the build was successful.

Reply:
Oh I think you mean why can't we use `--no-build` with `dotnet watch`. Not sure how useful `watch` would be if you can't build :)